### ISSUE TITLE
[1][kapeer42] Include changes on 'Changed/Deleted Proposal' in history

### ISSUE THREAD

**Issue by startailcoon** [bitshares/bitshares-ui#1864]
Issue #1673 included the ability to change/delete a proposed transaction. The conversation of the PR #1853 created work out of scope of that issue.

This issue is for hour tracking for changes mentioned in the PR comment https://github.com/bitshares/bitshares-ui/pull/1853#issuecomment-421643371

#### startailcoon commented
> I think an update should write out the new proposal object, as you can't see what have changed.
The delete may not require the object to be printed out, but could be of interest as the original proposal may be so old it's not visible in the history of transactions.

#### abitmore commented
> I think the description should be something like this:
aaa (read: the fee payer) updated proposal #xxx: 
added owner approval of aaa (read: the account in owner_approvals_to_add);
removed active approval of bbb (read: the account in active_approvals_to_remove);
added key approval of BTS7xxxxxxPub (read: the key in key_approvals_toadd);
etc.
As to the object, since it's not in the operation, it's unable to be printed in history.

**Comment by kapeer42** [comment]
@startailcoon 
> The delete may not require the object to be printed out, but could be of interest

The proposal object being deleted, I think that the only way to get the proposal content after deletion is to find the `proposal_create` operation which, to my knowledge, is only listed in the account history. 
This seems to contradict why you want the description in the `proposal_delete` in the first place.

Even if we did go through all the transaction in the account history to find the right `proposal_create` operation, I think there might be cases where the `proposal_delete` might appear in an account history that does not have the `proposal_create` in it, in that case I have no idea how I could find the proposal content except by going through all blocks in the chain.

I know this is counter-intuitive to the fact that the proposal content is displayed near the `proposal_delete` on screenshots I posted.

Otherwise, the update description part is almost done.
